### PR TITLE
chore: Single project - Counter App

### PR DESCRIPTION
**Pull request by vatsashah45**

### CODE REVIEW

**Review by agneszitte — COMMENTED** [review]
@vatsashah45, @eriklimakc 
Please add:
- A proper readme file with info + screenshots examples for the counter sample in the reference/Counter folder 
- Add the Counter sample in the list of samples in the global readme of the repo

**Comment by eriklimakc** [comment]
> @vatsashah45, @eriklimakc Please add:
> 
> * A proper readme file with info + screenshots examples for the counter sample in the reference/Counter folder
> * Add the Counter sample in the list of samples in the global readme of the repo

@vatsashah45 we can do something similar to what was done to [ChatGPT Sample](https://github.com/unoplatform/Uno.Samples/tree/master/UI/ChatGPT).

cc @agneszitte

File: reference/Counter/CSharp-MVVM/Counter/Counter.csproj [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<Project Sdk="Uno.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>
+      $(DotNetVersion)-browserwasm;
+      $(DotNetVersion)-android;
+      $(DotNetVersion)-ios;
+      $(DotNetVersion)-maccatalyst;
+      $(DotNetVersion)-windows10.0.19041;
+      $(DotNetVersion)-desktop;
+    </TargetFrameworks>
```
Comment (eriklimakc) [inline reference/Counter/CSharp-MVVM/Counter/Counter.csproj:10]:
```suggestion
    <TargetFrameworks>
      net8.0-desktop;
      net8.0-android;
      net8.0-ios;
      net8.0-maccatalyst;
      net8.0-windows10.0.19041;
      net8.0-browserwasm;
    </TargetFrameworks>
```

File: reference/Counter/CSharp-MVVM/Counter/Counter.csproj [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    <OutputType>Exe</OutputType>
+    <SingleProject>true</SingleProject>
```
Comment (eriklimakc) [inline reference/Counter/CSharp-MVVM/Counter/Counter.csproj:13]:
```suggestion
    <UnoSingleProject>true</UnoSingleProject>
```

File: reference/Counter/CSharp-MVUX/global.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  // To update the version of Uno please update the version of the Uno.Sdk here. See https://aka.platform.uno/upgrade-uno-packages for more information.
+  "msbuild-sdks": {
+    "Uno.Sdk": "5.2.0-dev.2103"
```
Comment (eriklimakc) [inline reference/Counter/CSharp-MVUX/global.json:4]: Update Uno.Sdk.

File: reference/Counter/CSharp-MVUX/Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<Project>
+  <PropertyGroup>
+    <DotNetVersion>net8.0</DotNetVersion>
```
Comment (eriklimakc) [inline reference/Counter/CSharp-MVUX/Directory.Build.props:3]:
```suggestion
```

File: reference/Counter/CSharp-MVUX/Directory.Build.props [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    -->
+    <NoWarn>$(NoWarn);NU1507;NETSDK1201;PRI257</NoWarn>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <UnoExtensionsVersion>4.1.0-dev.72</UnoExtensionsVersion>
+    <UnoToolkitVersion>5.2.0-dev.44</UnoToolkitVersion>
+    <UnoThemesVersion>4.2.0-dev.19</UnoThemesVersion>
+    <UnoCSharpMarkupVersion>5.2.0-dev.58</UnoCSharpMarkupVersion>
+  </PropertyGroup>
```
Comment (eriklimakc) [inline reference/Counter/CSharp-MVUX/Directory.Build.props:22]: Update uno packages.

File: reference/Counter/CSharp-MVUX/Counter/Counter.csproj [hunk 6]
```diff
@@ -0,0 +1,52 @@
+<Project Sdk="Uno.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>
+      $(DotNetVersion)-browserwasm;
+      $(DotNetVersion)-android;
+      $(DotNetVersion)-ios;
+      $(DotNetVersion)-maccatalyst;
+      $(DotNetVersion)-windows10.0.19041;
+      $(DotNetVersion)-desktop;
+    </TargetFrameworks>
```
Comment (eriklimakc) [inline reference/Counter/CSharp-MVUX/Counter/Counter.csproj:10]:
```suggestion
    <TargetFrameworks>
      net8.0-desktop;
      net8.0-android;
      net8.0-ios;
      net8.0-maccatalyst;
      net8.0-windows10.0.19041;
      net8.0-browserwasm;
    </TargetFrameworks>
```

File: reference/Counter/CSharp-MVUX/Counter/Counter.csproj [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+    <OutputType>Exe</OutputType>
+    <SingleProject>true</SingleProject>
```
Comment (eriklimakc) [inline reference/Counter/CSharp-MVUX/Counter/Counter.csproj:13]:
```suggestion
    <UnoSingleProject>true</UnoSingleProject>
```

File: reference/Counter/XAML-MVVM/Directory.Build.props [hunk 8]
```diff
@@ -0,0 +1,23 @@
+<Project>
+  <PropertyGroup>
+    <DotNetVersion>net8.0</DotNetVersion>
```
Comment (eriklimakc) [inline reference/Counter/XAML-MVVM/Directory.Build.props:3]:
```suggestion
```

File: reference/Counter/XAML-MVVM/Directory.Build.props [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    -->
+    <NoWarn>$(NoWarn);NU1507;NETSDK1201;PRI257</NoWarn>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <UnoExtensionsVersion>4.1.0-dev.72</UnoExtensionsVersion>
+    <UnoToolkitVersion>5.2.0-dev.44</UnoToolkitVersion>
+    <UnoThemesVersion>4.2.0-dev.19</UnoThemesVersion>
+    <UnoCSharpMarkupVersion>5.2.0-dev.58</UnoCSharpMarkupVersion>
+  </PropertyGroup>
```
Comment (eriklimakc) [inline reference/Counter/XAML-MVVM/Directory.Build.props:22]: Update uno packages.

File: reference/Counter/XAML-MVVM/global.json [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
+  // To update the version of Uno please update the version of the Uno.Sdk here. See https://aka.platform.uno/upgrade-uno-packages for more information.
+  "msbuild-sdks": {
+    "Uno.Sdk": "5.2.0-dev.2103"
```
Comment (eriklimakc) [inline reference/Counter/XAML-MVVM/global.json:4]: Update Uno.Sdk.

File: reference/Counter/XAML-MVVM/Counter/Counter.csproj [hunk 11]
```diff
@@ -0,0 +1,51 @@
+<Project Sdk="Uno.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>
+      $(DotNetVersion)-browserwasm;
+      $(DotNetVersion)-android;
+      $(DotNetVersion)-ios;
+      $(DotNetVersion)-maccatalyst;
+      $(DotNetVersion)-windows10.0.19041;
+      $(DotNetVersion)-desktop;
+    </TargetFrameworks>
```
Comment (eriklimakc) [inline reference/Counter/XAML-MVVM/Counter/Counter.csproj:10]:
```suggestion
    <TargetFrameworks>
      net8.0-desktop;
      net8.0-android;
      net8.0-ios;
      net8.0-maccatalyst;
      net8.0-windows10.0.19041;
      net8.0-browserwasm;
    </TargetFrameworks>
```

File: reference/Counter/XAML-MVVM/Counter/Counter.csproj [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+    <OutputType>Exe</OutputType>
+    <SingleProject>true</SingleProject>
```
Comment (eriklimakc) [inline reference/Counter/XAML-MVVM/Counter/Counter.csproj:13]:
```suggestion
    <UnoSingleProject>true</UnoSingleProject>
```

File: reference/Counter/XAML-MVUX/global.json [hunk 13]
```diff
@@ -0,0 +1,6 @@
+{
+  // To update the version of Uno please update the version of the Uno.Sdk here. See https://aka.platform.uno/upgrade-uno-packages for more information.
+  "msbuild-sdks": {
+    "Uno.Sdk": "5.2.0-dev.2103"
```
Comment (eriklimakc) [inline reference/Counter/XAML-MVUX/global.json:4]: Update Uno.Sdk.

File: reference/Counter/XAML-MVUX/Directory.Build.props [hunk 14]
```diff
@@ -0,0 +1,23 @@
+<Project>
+  <PropertyGroup>
+    <DotNetVersion>net8.0</DotNetVersion>
```
Comment (eriklimakc) [inline reference/Counter/XAML-MVUX/Directory.Build.props:3]:
```suggestion
```

File: reference/Counter/XAML-MVUX/Directory.Build.props [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+    -->
+    <NoWarn>$(NoWarn);NU1507;NETSDK1201;PRI257</NoWarn>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <UnoExtensionsVersion>4.1.0-dev.72</UnoExtensionsVersion>
+    <UnoToolkitVersion>5.2.0-dev.44</UnoToolkitVersion>
+    <UnoThemesVersion>4.2.0-dev.19</UnoThemesVersion>
+    <UnoCSharpMarkupVersion>5.2.0-dev.58</UnoCSharpMarkupVersion>
+  </PropertyGroup>
```
Comment (eriklimakc) [inline reference/Counter/XAML-MVUX/Directory.Build.props:22]: Update uno packages.

File: reference/Counter/XAML-MVUX/Counter/Counter.csproj [hunk 16]
```diff
@@ -0,0 +1,49 @@
+<Project Sdk="Uno.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>
+      $(DotNetVersion)-browserwasm;
+      $(DotNetVersion)-android;
+      $(DotNetVersion)-ios;
+      $(DotNetVersion)-maccatalyst;
+      $(DotNetVersion)-windows10.0.19041;
+      $(DotNetVersion)-desktop;
+    </TargetFrameworks>
```
Comment (eriklimakc) [inline reference/Counter/XAML-MVUX/Counter/Counter.csproj:10]:
```suggestion
    <TargetFrameworks>
      net8.0-desktop;
      net8.0-android;
      net8.0-ios;
      net8.0-maccatalyst;
      net8.0-windows10.0.19041;
      net8.0-browserwasm;
    </TargetFrameworks>
```

File: reference/Counter/XAML-MVUX/Counter/Counter.csproj [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+
+    <OutputType>Exe</OutputType>
+    <SingleProject>true</SingleProject>
```
Comment (eriklimakc) [inline reference/Counter/XAML-MVUX/Counter/Counter.csproj:13]:
```suggestion
    <UnoSingleProject>true</UnoSingleProject>
```

File: reference/Counter/CSharp-MVVM/global.json [hunk 18]
```diff
@@ -0,0 +1,6 @@
+{
+  // To update the version of Uno please update the version of the Uno.Sdk here. See https://aka.platform.uno/upgrade-uno-packages for more information.
+  "msbuild-sdks": {
+    "Uno.Sdk": "5.2.0-dev.2103"
```
Comment (eriklimakc) [inline reference/Counter/CSharp-MVVM/global.json:4]: Update Uno.Sdk.

File: reference/Counter/CSharp-MVVM/Directory.Build.props [hunk 19]
```diff
@@ -0,0 +1,23 @@
+<Project>
+  <PropertyGroup>
+    <DotNetVersion>net8.0</DotNetVersion>
```
Comment (eriklimakc) [inline reference/Counter/CSharp-MVVM/Directory.Build.props:3]:
```suggestion
```

File: reference/Counter/CSharp-MVVM/Directory.Build.props [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+    -->
+    <NoWarn>$(NoWarn);NU1507;NETSDK1201;PRI257</NoWarn>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <UnoExtensionsVersion>4.1.0-dev.72</UnoExtensionsVersion>
+    <UnoToolkitVersion>5.2.0-dev.44</UnoToolkitVersion>
+    <UnoThemesVersion>4.2.0-dev.19</UnoThemesVersion>
+    <UnoCSharpMarkupVersion>5.2.0-dev.58</UnoCSharpMarkupVersion>
+  </PropertyGroup>
```
Comment (eriklimakc) [inline reference/Counter/CSharp-MVVM/Directory.Build.props:22]: Update uno packages.

**Review by eriklimakc — CHANGES_REQUESTED** [review]

**Review by eriklimakc — COMMENTED** [review]
Looks great. Can we just move the solution content from, say, 'CSharp-MVUX/Counter' to 'CSharp-MVUX/'? We don't need a 'Counter' directory within the 4 versions.

**Review by eriklimakc — APPROVED** [review]

**Comment by agneszitte** [comment]
Closing this PR as the work will be done with https://github.com/unoplatform/Uno.Samples/pull/669

**Pull request closed without merging**